src/cli/cli.cpp

```diff
@@ -1918,24 +1921,75 @@ void Interpreter::HandleLinkMetricsReport(const otIp6Address *aAddress, const ot
     OutputIp6Address(*aAddress);
     OutputLine("");
 
-    if (aMetricsValues->mMetrics.mPduCount)
+    if (aMetricsValues != nullptr)
     {
-        OutputLine(" - PDU Counter: %d %s", aMetricsValues->mPduCountValue, kLinkMetricsTypeCount);
-    }
+        if (aMetricsValues->mMetrics.mPduCount)
+        {
+            OutputLine(" - PDU Counter: %d %s", aMetricsValues->mPduCountValue, kLinkMetricsTypeCount);
+        }
 
-    if (aMetricsValues->mMetrics.mLqi)
-    {
-        OutputLine(" - LQI: %d %s", aMetricsValues->mLqiValue, kLinkMetricsTypeAverage);
-    }
+        if (aMetricsValues->mMetrics.mLqi)
+        {
+            OutputLine(" - LQI: %d %s", aMetricsValues->mLqiValue, kLinkMetricsTypeAverage);
+        }
+
+        if (aMetricsValues->mMetrics.mLinkMargin)
+        {
+            OutputLine(" - Margin: %d (dB) %s", aMetricsValues->mLinkMarginValue, kLinkMetricsTypeAverage);
+        }
 
-    if (aMetricsValues->mMetrics.mLinkMargin)
+        if (aMetricsValues->mMetrics.mRssi)
+        {
+            OutputLine(" - RSSI: %d (dBm) %s", aMetricsValues->mRssiValue, kLinkMetricsTypeAverage);
+        }
+    }
+    else
     {
-        OutputLine(" - Margin: %d (dB) %s", aMetricsValues->mLinkMarginValue, kLinkMetricsTypeAverage);
+        OutputFormat("Link Metrics Report, status: ");
+        OutputLinkMetricsStatus(aStatus);
     }
+}
+
+void Interpreter::HandleLinkMetricsMgmtResponse(const otIp6Address *aAddress, uint8_t aStatus, void *aContext)
+{
+    static_cast<Interpreter *>(aContext)->HandleLinkMetricsMgmtResponse(aAddress, aStatus);
+}
+
+void Interpreter::HandleLinkMetricsMgmtResponse(const otIp6Address *aAddress, uint8_t aStatus)
+{
+    OutputFormat("Received Link Metrics Management Response from: ");
+    OutputIp6Address(*aAddress);
+    OutputLine("");
+
+    OutputFormat("Status: ");
+    OutputLinkMetricsStatus(aStatus);
+}
 
-    if (aMetricsValues->mMetrics.mRssi)
+void Interpreter::OutputLinkMetricsStatus(uint8_t aStatus)
+{
+    switch (aStatus)
     {
-        OutputLine(" - RSSI: %d (dBm) %s", aMetricsValues->mRssiValue, kLinkMetricsTypeAverage);
+    case OT_LINK_METRICS_STATUS_SUCCESS:
+        OutputLine("Success");
+        break;
+    case OT_LINK_METRICS_STATUS_CANNOT_SUPPORT_NEW_SERIES:
+        OutputLine("Cannot support new series");
+        break;
+    case OT_LINK_METRICS_STATUS_SERIESID_ALREADY_REGISTERED:
+        OutputLine("Series ID already registered");
+        break;
+    case OT_LINK_METRICS_STATUS_SERIESID_NOT_RECOGNIZED:
+        OutputLine("Series ID not recognized");
+        break;
+    case OT_LINK_METRICS_STATUS_NO_MATCHING_FRAMES_RECEIVED:
+        OutputLine("No matching series ID");
+        break;
+    case OT_LINK_METRICS_STATUS_OTHER_ERROR:
+        OutputLine("Other error");
+        break;
+    default:
+        OutputLine("Unknown error");
```

Comment:
nit: does having a single `OutputLine()` call after the switch statement reduce code size?

Reply:
Updated to use a `LinkMetricsStatusToStr` method to get static string and `OutputLine` with it